wgpu-core/src/track/texture.rs

```diff
@@ -118,14 +117,12 @@ impl ResourceState for TextureState {
             .enumerate()
         {
             let level = selector.levels.start + mip_id as hal::image::Level;
-            for &mut (aspect, ref mut plane_states) in &mut [
-                (hal::format::Aspects::COLOR, &mut mip.color),
-                (hal::format::Aspects::DEPTH, &mut mip.depth),
-                (hal::format::Aspects::STENCIL, &mut mip.stencil),
-            ] {
-                if !selector.aspects.contains(aspect) {
+            for &mut (mip_aspects, ref mut plane_states) in mip {
+                let aspects = selector.aspects & mip_aspects;
+                if aspects.is_empty() {
                     continue;
                 }
+                debug_assert_eq!(aspects, mip_aspects);
```

Comment:
Why do we always expect this to be the case? Or is it just unhandled for now?

Reply:
We do expect this: the unit of tracking needs to be always smaller (or equal) than the requested set of aspects:
  - if either is `color`, there is no other way for the other to also be `color` (there aren't formats with both color and non-color aspectrs)
  - if the unit is a merged depth-stencil, we require the calling side (tracking code) to always request them both (see the change to `texture_create_view`)
  - if the unit is separate depth-stencil, then each is always going to be a subset of the requested set of aspects

Reply:
Alright this makes sense, thanks!